wasm/main.py

```diff
@@ -2811,40 +2801,39 @@ def spec_externtype_matching_limits(limits_a: Limits, limits_b: Limits) -> str:
 def spec_externtype_matching(externtype1, externtype2):
     logger.debug('spec_externtype_matching(%s, %s)', externtype1, externtype2)
 
-    if externtype1[0] != externtype2[0]:
-        raise Unlinkable(f"Mismatch in extern types: {externtype1[0]} != {externtype2[0]}")
-    elif "func" == externtype1[0] and "func" == externtype2[0]:
-        if externtype1[1] == externtype2[1]:
+    if type(externtype1) is not type(externtype2):
+        raise Unlinkable(
+            f"Mismatch in extern types: {type(externtype1)} != {type(externtype2)}"
+        )
+    elif isinstance(externtype1, FuncType) and isinstance(externtype2, FuncType):
+        if externtype1 == externtype2:
             return "<="
         else:
-            raise Unlinkable(f"Function extern type mismatch: {externtype1[1]} != {externtype2[1]}")
-    elif "table" == externtype1[0] and "table" == externtype2[0]:
-        table_type_a, table_type_b = externtype1[1], externtype2[1]
+            raise Unlinkable(f"Function types not equal: {externtype1} != {externtype2}")
+    elif isinstance(externtype1, TableType) and isinstance(externtype2, TableType):
+        spec_externtype_matching_limits(externtype1.limits, externtype2.limits)
 
-        limits_a, limits_b = table_type_a.limits, table_type_b.limits
-        spec_externtype_matching_limits(limits_a, limits_b)
-
-        elem_type_a, elem_type_b = table_type_a.elem_type, table_type_b.elem_type
-        if elem_type_a is elem_type_b:
+        if externtype1.elem_type is externtype1.elem_type:
```

Comment:
It seems like this one should be an `==`, since another instance of the `NamedTuple` should match if all the values match (I would think).

Also, I assume the second variable name is a typo.
```suggestion
        if externtype1.elem_type == externtype2.elem_type:
```

Reply:
`elem_type` is a class, not an instance so defaulted to `is` check (see other comment for why).

Reply:
:+1: yeah, somehow I thought it was an instance :man_facepalming: